packages/editor/src/browser/editor-widget.ts

```diff
@@ -18,6 +18,11 @@ import { SelectionService } from '@theia/core/lib/common';
 import { Widget, BaseWidget, Message, Saveable, SaveableSource, Navigatable } from '@theia/core/lib/browser';
 import URI from '@theia/core/lib/common/uri';
 import { TextEditor } from './editor';
+import { SplitPanel } from '@phosphor/widgets';
+
+export abstract class EditorBasedSplitPanel extends SplitPanel {
```

Comment:
This should be an interface independent of the SplitPanel:
```
export interface EditorWidgetProvider {
    getEditor(): EditorWidget;
}
export namespace EditorWidgetProvider {
    export function is(provider: object | undefined): provider is EditorWidgetProvider {
        return !!provider && 'getEditor' in provider;
    }
}
```

Reply:
@svenefftinge Done!